component/file_cache/file_cache.go

```diff
@@ -691,69 +696,85 @@ func (fc *FileCache) DeleteFile(options internal.DeleteFileOptions) error {
 	return nil
 }
 
-// OpenFile: Makes the file available in the local cache for further file operations.
-func (fc *FileCache) OpenFile(options internal.OpenFileOptions) (*handlemap.Handle, error) {
-	log.Trace("FileCache::OpenFile : name=%s, flags=%d, mode=%s", options.Name, options.Flags, options.Mode)
+func (fc *FileCache) downloadFile(handle *handlemap.Handle) (*handlemap.Handle, error) {
 
-	localPath := common.JoinUnixFilepath(fc.tmpPath, options.Name)
+	//extract flags and mode out of the value from handle
+	var flags int
+	var fMode fs.FileMode
+
+	flagMode, found := handle.GetValue("fileFlagMode")
+	if found {
+		fileOptions, ok := flagMode.(openFileOptions)
+		if !ok {
+			log.Err("FileCache::DownloadFile : error Type assertion failed on getting flag for %s", handle.Path)
+			return handle, fmt.Errorf("type assertion failed on getting flag for %s", handle.Path)
+		}
```

Comment:
I think this type assertion should follow the pattern set in other places where `GetValue` is used. Like in libfuse2_handler.go.